corehq/apps/app_manager/dbaccessors.py

```diff
@@ -324,3 +324,8 @@ def get_case_types_from_apps(domain):
          .size(0)
          .terms_aggregation('modules.case_type.exact', 'case_types'))
     return set(q.run().aggregations.case_types.keys)
+
+
+def get_case_sharing_apps_in_domain(domain, exclude_app_id=None):
```

Comment:
this seems like an expensive call - have you tested on domains with lots of apps? if it is called frequently (every time a form opens?) is there any caching we could add? i think you would have to cache just the case types so it wasn't huge